packages/amazonq/src/app/inline/sessionManager.ts

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { InlineCompletionItemWithReferences } from '@aws/language-server-runtimes-types/inlineCompletionWithReferences'
+import { InlineCompletionItem } from 'vscode'
+
+// TODO: add more needed data to the session interface
+interface CodeWhispererSession {
+    sessionId: string
+    suggestions: InlineCompletionItemWithReferences[]
+    // TODO: might need to convert to enum states
+    isRequestInProgress: boolean
+    firstCompletionDisplayLatency?: number
+}
+
+export class SessionManager {
+    static #instance: SessionManager
```

Comment:
same comment about instances with this one. Can we get rid of the instance and just pass it through a constructor?